Basic-Car-Maintenance/Shared/MainView/ViewModels/MainTabViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MainTabViewModel.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Omar Hegazy on 13/10/2023.
+//
+
+import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import SwiftData
+
+@Observable
+class MainTabViewModel {
+    private var acknowledgedAlerts = [AcknowledgedAlert]()
+    @MainActor var alert: AlertItem?
+    let modelContext: ModelContext
+    
+    init(modelContext: ModelContext) {
+        self.modelContext = modelContext
+        fetchAcknowledgedAlerts()
+    }
+    
+    /// Update the UI once a new alert is sent
+    func listenToAlertsUpdates() {
+        var query = Firestore
+            .firestore()
+            .collection("alerts")
+            .whereField("isOn", isEqualTo: true)
+            .limit(to: 1)
+        
+        if !acknowledgedAlerts.isEmpty {
+            query = query
+                .whereField("_id", notIn: self.acknowledgedAlerts.map(\.id) as [String])
```

Comment:
Why does this need to be `_id` rather than just `id`?

Reply:
Gotcha, also btw I'll update the original issue, but I just changed the model so it's not a UUID, but it will be the same as the document ID (because looking at it, it makes sense to just use the same ID, because it will be unique)

<img width="464" alt="Screenshot 2023-10-16 at 11 27 27 PM" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/839f6499-7431-4463-a2a9-719a58756475">